downloads/swan-lake-release-notes/swan-lake-preview3/RELEASE_NOTE.md

```diff
@@ -2,29 +2,279 @@
+    if (!(resultStr is ())) {
+        sql:Error? e = resultStr.forEach(function(record{} result) {
+        io:println("Full Customer details: ", result);
+      });
+    } else {
+        io:println("Stored  procedure does not return anything.");
+    }
+}
+
+```
+##### Module Organization
+
+The `ballerina/nats` library was moved to Ballerina Central. Previously, this module was packed in the Ballerina distribution. With this change, this library can now be released independently.
+
+#### Developer Tools
+
+##### Language Server
+
+###### Introducing AI-based `Data Mapping` Code Action
+
+Two record types can now be mapped automatically using the `Data Mapping` code action. Once a possible record mapping instance is identified, it suggests a mapping based on an AI algorithm. A mapping function will be generated automatically and added to the workspace to perform the record mapping.
+
+The following is a sample in which the code action to generate a mapping function will appear when attempting to assign a mapping value to a variable of a type that is not directly assignable. 
+
+```ballerina
+type Grade record {|
+   int maths;
+   int physics;
+   int chemistry;
+|};
+ 
+type NameAndGrade record {|
+   string name;
+   string surname;
+   int maths;
+   int physics;
+   int chemistry;
+|};
+ 
+public function main() {
+   NameAndGrade student = {
+	name: "Kamal",
+	surname: “Perera”,
+	maths: 90,
+physics: 99,
+chemistry: 95
+   };
+   Grade grades = student;
+}
+```
+By choosing the `Generate mapping function`code action, the following function will be added to the workspace.
+
+```ballerina
+function mapNameAndGradeToGrade(NameAndGrade nameAndGrade) returns Grade {
+// Some record fields might be missing in the AI based mapping.
+   Grade grade = {
+maths: nameAndGrade.maths, 
+physics: nameAndGrade.physics, 
+chemistry: nameAndGrade.chemistry};
+   return grade;
+}
+```
+
+Furthermore, the line with the error would be replaced with a function call as shown below.
+
+```ballerina
+Grade grades = mapNameAndGradeToGrade(student);
+```
+
+For more information, see [Code Actions](https://ballerina.io/swan-lake/learn/setting-up-visual-studio-code/language-intelligence/#code-actions).
+
+##### Test Framework
+
+###### Support single test execution
+
+A single test function or a set of functions can now be executed using the `--tests` flag as follows.
+
+```cmd
+$ ballerina test --tests <test_function> --all
+```
+
+###### API change in `assertEquals` and `assertNotEquals` functions
+
+Deep value equality is supported only for `anydata`-typed values according to the language specification. The `assertEquals` function has been changed to accept only  `anydata`-typed values to reflect this behavior.
+
+###### Introduction of the `assertExactEquals` and `assertNotExactEquals` functions
+
+The `assertExactEquals` function compares two values to assert whether they refer to the same entity (i.e., they are exactly equal). 
+
+
+**Example:**
```

Comment:
For some examples, we have this line. But not for all the examples. Can we be consistent on adding this?

Reply:
Fixed